### PR TITLE
Squash all PRs in Prow by default

### PR DESCRIPTION
**Pull request by RedbackThomson**
Fixes the config to specify to squash all ACK repository PRs

### CODE REVIEW

**Review by jaypipes — CHANGES_REQUESTED** [review]
Although I am a big proponent of `git rebase`'ing and squashing sausagemaking commits into properly-formatted and independently-testable commits, I'm not a supporter of auto-squash-merging. The reason is because the commit message for the squash-merged PRs is just a concatenation of all of the sausagemaking commit messages, which isn't useful and is exactly the thing we want to get rid of by asking contributors to `git rebase` and squash their commits into independently-testable and well-documented (commit message-wise) commits.

In other words, I don't want to auto-squash-merge PRs and end up with a bunch of commit messages that look like this:

```
* chore: migrate repo to new format
* clean up code
* revert previous commit
* add to tests
* Update AWS SDK Go
* Go mod tidy
* fix: missing formatting
* samples added
```

**Comment by RedbackThomson** [comment]
Found a way to template the squashed commits with the commit message as the title of the PR and the commit body as the description of the PR.

**Review by jaypipes — APPROVED** [review]
:+1:

**Comment by A-Hilaly** [comment]
>Found a way to template the squashed commits with the commit message as the title of the PR and the commit body as the description of the PR

Does this mean that preferably we should have one commit per PR, or at least a a PR description containing summary of all the commits (if there is more than one) ?

**Comment by RedbackThomson** [comment]
> > Found a way to template the squashed commits with the commit message as the title of the PR and the commit body as the description of the PR
> 
> Does this mean that preferably we should have one commit per PR, or at least a a PR description containing summary of all the commits (if there is more than one) ? @RedbackThomson

This means that whatever the description of the PR is, that will be the body of the commit. I think this is preferable because typically the body of a PR gives more contexts and provides a better summary than any number of commits could ever provide.

**Comment by A-Hilaly** [comment]
I'm kinda worried about the current template (using PR body)... look the [latest commit here](https://github.com/aws-controllers-k8s/dev-tools/commits/main) it's starting with "issue N/A" , at one point we have `By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.` then the commit message...

**Comment by A-Hilaly** [comment]
Most of the developers are not aware of the new merge type and they will probably focus more on commit message than the PR body it self (https://github.com/aws-controllers-k8s/code-generator/pull/69#pullrequestreview-663996218).
I don't think that we should explain this to every current and future contributors. IMO i believe we should go back to "merge" method which more obvious and expected. @RedbackThomson

**Comment by jaypipes** [comment]
> I'm kinda worried about the current template (using PR body)... look the [latest commit here](https://github.com/aws-controllers-k8s/dev-tools/commits/main) it's starting with "issue N/A" , at one point we have `By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.` then the commit message...

Yeah, that definitely was not the intention. That latest commit should have had `github: find users personal fork using ListRepositoryForks` as the commit message summary (since that is the [title of the PR](https://github.com/aws-controllers-k8s/dev-tools/pull/16) from which the commit was squash-merged).

The body of the commit message should have been the description of the PR, so this:

```
Issue: N/A

Recently Github API /user/{user}/repos endpoint changed and stopped returning information about repositories parents (repos from which they were forked) and started to cause some nil pointer panics in ensure repo command (which not in main branch yet).
This patch uses ListRepositoryForks in GetUserRepositoryFork to find a user personal fork. We could could have used GetRepository call to read the parent repository information, but it this approach will require way more API requests.

Description of changes:

    Call ListRepositoryForks in GetUserRepositoryFork to find a user personal fork.
    upgrade go-github version to v35
    regenerate pkg/github mocks
    renames ErrorForkNotFound to ErrForkNotFound

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
```

I'm less concerned about the "By submitting this pull request.." wording being included in the commit message, but definitely agree the title and body content should not look like it does.

File: prow/config/templates/config-ConfigMap.yaml [hunk 1]
```diff
@@ -83,8 +83,14 @@ data:
             sidecar: {{ .Values.utility_images.sidecar }}
 
     tide:
+      merge_commit_template:
+        "{{ .Values.github.organisation }}":
+          body: >-
+              {{ "{{ .Title }}" }}
+          title: >-
+              {{ "{{ .Body }}" }}
```
Comment (jaypipes) [inline prow/config/templates/config-ConfigMap.yaml:91]: @RedbackThomson @A-Hilaly LOL. Please see above. :) We have the body and title switched :(

Comment (A-Hilaly) [reply prow/config/templates/config-ConfigMap.yaml:91]: Ahh i think was fixed in https://github.com/aws-controllers-k8s/test-infra/pull/28

Comment (A-Hilaly) [reply prow/config/templates/config-ConfigMap.yaml:91]: Or maybe it should be something like this?
```
          title: >-
              {{ "{{ .Title }}" }}
          body: >-
              {{ "{{ .Body }}" }}
```

Comment (jaypipes) [reply prow/config/templates/config-ConfigMap.yaml:91]: > Or maybe it should be something like this?
> 
> ```
>           title: >-
>               {{ "{{ .Title }}" }}
>           body: >-
>               {{ "{{ .Body }}" }}
> ```

yes, this. :)
